### ISSUE TITLE
[core] Disable ShortVariable

### ISSUE THREAD

**Issue by mcandre** [pmd/pmd#1601]
**Affects PMD Version:** 

0.3+

**Rule:**

ShortVariable

**Description:**

When there is no additional information to report about a variable, such as a string parameter or loop index iterator or element, then it is both clear and convenient to express the variable with a short name such as a single character. See: https://en.wikipedia.org/wiki/Algebra

As a workaround, I am configuring this in my rulesets.xml:

```xml
<rule ref="category/java/codestyle.xml">
    <exclude name="ShortVariable" />
</rule>
```

**Comment by mcandre** [comment]
Anywhere aside from loops that I want to declare a variable as `<Class> <a letter>`.

**Comment by mcandre** [comment]
Yeah no, people have been using this naming convention for centuries, since the time of algebra. It's more tedious and dangerous to beg users to make up a more meaningful name for variables when the code in question is designed to be generic enough to take some *X*, *Y*, or *Z* thing and return a transformation on those otherwise unspecified parameters.

**Comment by adangel** [comment]
In that case, you want to exclude the rule for this area of code. Not all rules fit for every user and every project. Take PMD as an advice - sometimes it makes sense, sometimes not.

If you code lives in a package (e.g. com.example.math) and you are Ok with using Short Variables there, then you could disable the rule for this package (using excludes or suppression).
